packages/opensrp-plans/src/components/AssignmentTable/index.tsx

```diff
@@ -0,0 +1,195 @@
+reducerRegistry.register(jurisdictionReducerName, JurisdictionsReducer);
+
+export interface AssignmentTableProps extends CommonProps {
+  serviceClass: typeof OpenSRPService;
+  assignmentsActionCreator: typeof fetchAssignments;
+  organizationsActionCreator: typeof fetchOrganizations;
+  jursActionCreator: typeof fetchJurisdictions;
+  planCreator: typeof fetchPlanDefinitions;
+  assignments: Assignment[];
+  organizations: Organization[];
+  plan: PlanDefinition;
+  jurisdictions: Jurisdiction[];
+  assignAtGeoLevel: number;
+  disableAssignments: boolean;
+}
+
+const defaultProps = {
+  ...defaultCommonProps,
+  serviceClass: OpenSRPService,
+  assignmentsActionCreator: fetchAssignments,
+  organizationsActionCreator: fetchOrganizations,
+  jursActionCreator: fetchJurisdictions,
+  planCreator: fetchPlanDefinitions,
+  assignments: [],
+  organizations: [],
+  jurisdictions: [],
+  assignAtGeoLevel: 0,
+  disableAssignments: false,
+};
+
+/** Table component for eusm plan assignments, requires plan
+ *
+ * @param props -the props
+ */
+const AssignmentTable = (props: AssignmentTableProps) => {
+  const {
+    plan,
+    serviceClass,
+    assignments,
+    organizations,
+    assignmentsActionCreator,
+    organizationsActionCreator,
+    jursActionCreator,
+    jurisdictions,
+    baseURL,
+    planCreator,
+    assignAtGeoLevel,
+    disableAssignments,
+  } = props;
+  const { handleBrokenPage, broken, errorMessage } = useHandleBrokenPage();
+  const [loading, setLoading] = useState<boolean>(true);
+  const planId = plan.identifier;
+
+  // todo component should re-render make the calls to get
+  useEffect(() => {
+    const assignmentsPromise = loadAssignments(
+      baseURL,
+      planId,
+      serviceClass,
+      assignmentsActionCreator
+    );
+    const organizationsPromise = loadOrganizations(
+      baseURL,
+      serviceClass,
+      organizationsActionCreator
+    );
+    const jursPromise = loadJurisdictions(
+      baseURL,
+      assignAtGeoLevel,
+      jursActionCreator,
+      serviceClass
+    );
+
+    Promise.all([assignmentsPromise, organizationsPromise, jursPromise])
+      .catch((e) => {
+        handleBrokenPage(e);
+      })
+      .finally(() => {
+        setLoading(false);
+      });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [plan.jurisdiction]);
+
+  if (broken) {
+    return <BrokenPage errorMessage={errorMessage} />;
+  }
+
+  const planJurisdictions = plan.jurisdiction.map((jurCode) => jurCode.code);
+  const dataSource = getDataSource(organizations, jurisdictions, assignments, planJurisdictions);
+  const columns = getPlanAssignmentColumns(
+    assignments,
+    organizations,
+    jurisdictions,
+    serviceClass,
+    planCreator,
+    assignmentsActionCreator,
+    plan,
+    baseURL,
+    disableAssignments
+  );
+
+  return (
+    <div className="assignment-table">
+      <Table dataSource={dataSource} loading={loading} columns={columns} pagination={false}></Table>
```

Comment:
Are we are certain the table should always not have pagination?

Reply:
@ciremusyoka yeah, actually, for the MVP it only has a single row.